gtsam/hybrid/HybridGaussianFactorGraph.cpp

```diff
@@ -238,20 +235,14 @@ discreteElimination(const HybridGaussianFactorGraph &factors,
     } else if (auto gmf = dynamic_pointer_cast<HybridGaussianFactor>(f)) {
       // Case where we have a HybridGaussianFactor with no continuous keys.
       // In this case, compute discrete probabilities.
-      auto logProbability =
-          [&](const GaussianFactor::shared_ptr &factor) -> double {
-        // If the factor is null, it is has been pruned hence return ∞
-        // so that the exp(-∞)=0.
-        if (!factor) return std::numeric_limits<double>::infinity();
-        return factor->error(VectorValues());
+      auto potential = [&](const auto& pair) -> double {
+        auto [factor, scalar] = pair;
+        // If factor is null, it has been pruned, hence return potential zero
+        if (!factor) return 0.0;
+        return exp(-scalar - factor->error(kEmpty));
```

Comment:
Confirming that I am running into a similar issue as before.

Reply:
Ok, I'll fix this

Reply:
I will fix it *only* here, when the exp is taken. I'd like the original errors preserved in the factors, not shifted before it's needed. I can explain in person if you vehemently object.

Reply:
OK, for this case and case below, I added a function 
```c++
/// Take negative log-values, shift them so that the minimum value is 0, and
/// then exponentiate to create a DecisionTreeFactor (not normalized yet!).
static DecisionTreeFactor::shared_ptr DiscreteFactorFromErrors(
    const DiscreteKeys &discreteKeys,
    const AlgebraicDecisionTree<Key> &errors) {
  double min_log = errors.min();
  AlgebraicDecisionTree<Key> potentials = DecisionTree<Key, double>(
      errors, [&min_log](const double x) { return exp(-(x - min_log)); });
  return std::make_shared<DecisionTreeFactor>(discreteKeys, potentials);
}
```
Please note I realized it is *not* necessary to normalize to probabilities: it is still just a *factor* that we need, not a *conditional*. That was an error on my part before - although it only cost some computation, nothing else.

Reply:
I believe these are the only two places.

Also I should explain: the shifting is what we need, not necessarily the normalization by the sum, so I agree with this assessment.

I was going to make a PR once I figured out the Windows error, so I'll just rebase your changes on to my branch and run the experiments again.

Reply:
Also I would like to hear the explanation for my own curiosity. I have a suspicion we may need to normalize somewhere.

Reply:
Actually, it might still need to happen in the discretePosterior methods. It seems what we need is a constructor in DecisionTreeFactor. factors are not normalized, so for discretePosterior it should be a constructor in DiscreteConditional (which is normalized).

Reply:
I can do it and piggyback it off of #1867

Reply:
Apparently we can't create a DiscreteConditional since we don't have the cardinality information of the Keys from an AlgebraicDecisionTree. Thoughts?

Reply:
Also `discretePosterior` already normalizes due to the `p / p.sum()`.